templates/goshared/msg.go

```diff
@@ -51,9 +51,16 @@ func (m {{ (msgTyp .).Pointer }}) validate(all bool) error {
 			}
 		{{ end }}
 
+		{{ range .SyntheticOneOfFields }}
+			if m.{{ name . }} != nil {
+				{{ render (context .) }}
+			}
+		{{ end }}
+
```

Comment:
Do we need something similar to this for Java and C++?

Reply:
We do, though I haven't yet made changes to those templates. If someone wants to pick those up, that'd be appreciated otherwise i'll look into it next week as this week is a bit busy for me.

I've fixed the breaking changes at least though, which means for now the proto3 optional fields aren't technically supported unless `lang=go`.

I wanted to implement this logic where compile would fail if the user was using `proto3: optional` && `lang!=go` in the interim, but haven't yet found a way to do so cleanly. The params and PGS init are being done independent of each other IIRC.